### PR TITLE
[Pigeon] Add toString to the genereated class

### PR DESCRIPTION
**Pull request by SchabanBo**
This PR added the override for the toString method as mentioned in [Issue](https://github.com/flutter/flutter/issues/59027)

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

### CODE REVIEW

File: packages/pigeon/CHANGELOG.md
Comment (tarrinneal) [inline packages/pigeon/CHANGELOG.md:3]: Make sure you check out the style guide for plugins/packages. https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style

Comment (SchabanBo) [reply packages/pigeon/CHANGELOG.md]: I have adjusted this as the style guide requires.

**Review by tarrinneal — DISMISSED** [review]

**Comment by tarrinneal** [comment]
You will need to update this to the latest commit on main, and then we should be able to get it merged

**Comment by tarrinneal** [comment]
@gaaclarke Just wanted to wait for you to take a quick look at this pr before I merge it. Does this implementation of toString do what you think it should?

File: packages/pigeon/lib/dart_generator.dart
Comment (gaaclarke) [inline packages/pigeon/lib/dart_generator.dart:593]: Sorry, I don't think we should do this.  It is leaking information about the wireformat to the public API.  Remember we want to be able to completely isolate the wireformat from users so that we can switch it out without breaking people.

Comment (gaaclarke) [reply packages/pigeon/lib/dart_generator.dart:593]: If you still want a `toString` method it shouldn't rely on `encode()` and it it probably shouldn't output a format that is trivially parsed.

**Review by gaaclarke — CHANGES_REQUESTED** [review]

**Pull request closed without merging**
